anymaniax/orval
# feat(core): Only consider non-readonly properties in request bodies

## Status



**READY**

## Description

Whenever a readonly property is defined in a request body then these may be sent as part of a response but should not be sent as part of a request according to the openapi 3.0 specification. To solve this using types, we can only allow the properties that are non-readonly since these are the actual request body.

Should fix: #616

## Steps to Test or Reproduce

Outline the steps to test or reproduce the PR here.

1. Use a readonly property in a requestBody
2. Check that the readonly properties are discarded from the generated body type in the defined hooks.
